project/SparkBuild.scala

```diff
@@ -163,14 +163,15 @@ object SparkBuild extends PomBuild {
         val configUrlV = scalastyleConfigUrl.in(config).value
         val streamsV = streams.in(config).value
         val failOnErrorV = true
+        val failOnWarningV = false
```

Comment:
I disabled this as we did not do this before and, I found we actually looks replacing a case to `warn` on sbt compile and test - https://github.com/apache/spark/pull/19116/files#diff-c3580fe26fb42eb3aac6e180ae11e947R139  although every level in `scalastyle-config.xml` is `error`.

Reply:
Thanks @viirya, actually I found this while double checking `println`. FWIW, I double checked `import scala.Predef.{println => _, _}` still fails.

Reply:
I've been noticed this, but looks like we only have error level rules in `scalastyle-config.xml`. Not aware of a rewriting hidden there.